src/test/java/net/datafaker/service/WeightedRandomSelectorTest.java

```diff
@@ -0,0 +1,326 @@
+    void testWeightedArrayElement_withInvalidWeightType(WeightedRandomSelector weightedRandomSelector) {
+        List<Map<String, Object>> items = List.of(
+            Map.of(VALUE_KEY, ELEMENT_1, WEIGHT_KEY, STRING_WEIGHT),
+            Map.of(VALUE_KEY, ELEMENT_2, WEIGHT_KEY, ELEMENT_2_WEIGHT)
+        );
+
+        assertThatThrownBy(() -> weightedRandomSelector.select(items))
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessage("Weight must be a non-null Double");
+    }
+
+    @ParameterizedTest
+    @MethodSource("weightedRandomSelectorProvider")
+    void testWeightedArrayElement_withNaNWeight(WeightedRandomSelector weightedRandomSelector) {
+        List<Map<String, Object>> items = List.of(
+            Map.of(VALUE_KEY, ELEMENT_1, WEIGHT_KEY, Double.NaN),
+            Map.of(VALUE_KEY, ELEMENT_2, WEIGHT_KEY, ELEMENT_2_WEIGHT)
+        );
+
+        assertThatThrownBy(() -> weightedRandomSelector.select(items))
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessage("Weight must be a positive number and cannot be NaN or infinite");
+    }
+
+    @ParameterizedTest
+    @MethodSource("weightedRandomSelectorProvider")
+    void testWeightedArrayElement_withZeroWeight(WeightedRandomSelector weightedRandomSelector) {
+        List<Map<String, Object>> items = List.of(
+            Map.of(VALUE_KEY, ELEMENT_1, WEIGHT_KEY, ZERO_WEIGHT),
+            Map.of(VALUE_KEY, ELEMENT_2, WEIGHT_KEY, ELEMENT_2_WEIGHT)
+        );
+
+        assertThatThrownBy(() -> weightedRandomSelector.select(items))
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessage("Weight must be a positive number and cannot be NaN or infinite");
+    }
+
+    @ParameterizedTest
+    @MethodSource("weightedRandomSelectorProvider")
+    void testWeightedArrayElement_withNegativeWeight(WeightedRandomSelector weightedRandomSelector) {
+        List<Map<String, Object>> items = List.of(
+            Map.of(VALUE_KEY, ELEMENT_1, WEIGHT_KEY, NEGATIVE_WEIGHT),
+            Map.of(VALUE_KEY, ELEMENT_2, WEIGHT_KEY, ELEMENT_2_WEIGHT)
+        );
+
+        assertThatThrownBy(() -> weightedRandomSelector.select(items))
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessage("Weight must be a positive number and cannot be NaN or infinite");
+    }
+
+    @ParameterizedTest
+    @MethodSource("weightedRandomSelectorProvider")
+    void testWeightedArrayElement_withInfiniteWeight(WeightedRandomSelector weightedRandomSelector) {
+        List<Map<String, Object>> items = List.of(
+            Map.of(VALUE_KEY, ELEMENT_1, WEIGHT_KEY, Double.POSITIVE_INFINITY),
+            Map.of(VALUE_KEY, ELEMENT_2, WEIGHT_KEY, ELEMENT_2_WEIGHT)
+        );
+
+        assertThatThrownBy(() -> weightedRandomSelector.select(items))
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessage("Weight must be a positive number and cannot be NaN or infinite");
+    }
```

Comment:
It looks like all these tests (my be some more) could be replaced with one parameterized test
since we have input of key/value sequence, exception and error message

Reply:
Good point, I refactored a whole test class to use a parameterized tests.